src/controls/control.class.ts

```diff
@@ -1,372 +1,392 @@
-     * controls in the corner, that stay in the same position when the object scale.
-     * of the bounding box.
-     * @type {Number}
-     * @default 0
-     */
-    offsetX: 0,
+  /**
+   * Sets the height of the control. If null, defaults to object's cornerSize.
+   * Expects both sizeX and sizeY to be set when set.
+   * @type {?Number}
+   * @default null
+   */
+  sizeY: number | null = null;
 
-    /**
-     * Vertical offset of the control from the defined position. In pixels
-     * Positive offset moves the control to the bottom, negative to the top.
-     * @type {Number}
-     * @default 0
-     */
-    offsetY: 0,
+  /**
+   * Sets the length of the touch area of the control. If null, defaults to object's touchCornerSize.
+   * Expects both touchSizeX and touchSizeY to be set when set.
+   * @type {?Number}
+   * @default null
+   */
+  touchSizeX: number | null = null;
 
-    /**
-     * Sets the length of the control. If null, defaults to object's cornerSize.
-     * Expects both sizeX and sizeY to be set when set.
-     * @type {?Number}
-     * @default null
-     */
-    sizeX: null,
+  /**
+   * Sets the height of the touch area of the control. If null, defaults to object's touchCornerSize.
+   * Expects both touchSizeX and touchSizeY to be set when set.
+   * @type {?Number}
+   * @default null
+   */
+  touchSizeY: number | null = null;
 
-    /**
-     * Sets the height of the control. If null, defaults to object's cornerSize.
-     * Expects both sizeX and sizeY to be set when set.
-     * @type {?Number}
-     * @default null
-     */
-    sizeY: null,
+  /**
+   * Css cursor style to display when the control is hovered.
+   * if the method `cursorStyleHandler` is provided, this property is ignored.
+   * @type {String}
+   * @default 'crosshair'
+   */
+  cursorStyle = 'crosshair';
 
-    /**
-     * Sets the length of the touch area of the control. If null, defaults to object's touchCornerSize.
-     * Expects both touchSizeX and touchSizeY to be set when set.
-     * @type {?Number}
-     * @default null
-     */
-    touchSizeX: null,
+  /**
+   * If controls has an offsetY or offsetX, draw a line that connects
+   * the control to the bounding box
+   * @type {Boolean}
+   * @default false
+   */
+  withConnection = false;
 
-    /**
-     * Sets the height of the touch area of the control. If null, defaults to object's touchCornerSize.
-     * Expects both touchSizeX and touchSizeY to be set when set.
-     * @type {?Number}
-     * @default null
-     */
-    touchSizeY: null,
-
-    /**
-     * Css cursor style to display when the control is hovered.
-     * if the method `cursorStyleHandler` is provided, this property is ignored.
-     * @type {String}
-     * @default 'crosshair'
-     */
-    cursorStyle: 'crosshair',
+  constructor(options: Partial<Control>) {
+    for (const key in options) {
+      // @ts-expect-error TS fails to understand the assignment
+      this[key] = options[key];
+    }
```

Comment:
but why can't we have `Object.assign(this, options)` ? why do we need a loop?

Reply:
we can do that